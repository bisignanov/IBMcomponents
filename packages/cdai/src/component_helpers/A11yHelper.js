//
// Copyright IBM Corp. 2020, 2020
//
// This source code is licensed under the Apache-2.0 license found in the
// LICENSE file in the root directory of this source tree.
//

export const keys = {
  SPACE: ' ',
  TAB: 'Tab',
  CLICK: 'click',
  KEYPRESS: 'keypress',
  KEYDOWN: 'keydown',
  KEYUP: 'keyup',
  ENTER: 'Enter',
};

export const handlers = {
  onClick: 'onClick',
  onKeyDown: 'onKeyDown',
  onKeyUp: 'onKeyUp',
  onKeyPress: 'onKeyPress',
};

const defaultValidKeys = [keys.CLICK, keys.SPACE, keys.ENTER];
const defaultHandlers = [handlers.onClick, handlers.onKeyDown];

// triggers a callback if event key matches one in validKeys
export const keyAndClickHandler = (callback, validKeys = defaultValidKeys) => {
  return (evt) => {
    if (evt && (evt.type || evt.key)) {
      if (validKeys.includes(evt.type) || validKeys.includes(evt.key)) {
        callback(evt);
      }
    }
  };
};

export const createInteractionHandler = (
  callback,
  handlers = defaultHandlers,
  validKeys = defaultValidKeys
) => {
  let handler = {};

  for (let i = 0; i < handlers.length; i++) {
    handler[handlers[i]] = keyAndClickHandler(callback, validKeys);
  }
  return handler;
};
