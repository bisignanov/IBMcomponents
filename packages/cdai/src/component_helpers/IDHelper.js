//
// Copyright IBM Corp. 2020, 2020
//
// This source code is licensed under the Apache-2.0 license found in the
// LICENSE file in the root directory of this source tree.
//

const attribute = 'data-ui-id';

export function idAttributeSelector(identifier) {
  let attr = `[${attribute}="${identifier}"]`;
  return attr;
}

export function fuzzyAttributeSelector(identifier) {
  let attr = `[${attribute}*="${identifier}"]`;
  return attr;
}

export function idAttribute(identifier) {
  let attr = { [attribute]: identifier };
  return attr;
}
