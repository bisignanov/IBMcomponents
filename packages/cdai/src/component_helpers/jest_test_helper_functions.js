//
// Copyright IBM Corp. 2020, 2020
//
// This source code is licensed under the Apache-2.0 license found in the
// LICENSE file in the root directory of this source tree.
//
import React from 'react';
import { shallow, mount } from 'enzyme';

export const getJSXForComponent = (
  component,
  defaultProps = {},
  customProps = {}
) => {
  return React.createElement(component, { ...defaultProps, ...customProps });
};

export const mountComponent = (jsx, returnShallow = true) => {
  let component, componentInstance;

  if (returnShallow) {
    component = shallow(jsx);
  } else {
    component = mount(jsx);
  }

  componentInstance = component.instance();

  let unmount = () => {
    unmountComponent(component);
  };

  return {
    component,
    componentInstance,
    unmount,
  };
};

export const unmountComponent = (component) => {
  component && component.length > 0 && component.unmount();
};
