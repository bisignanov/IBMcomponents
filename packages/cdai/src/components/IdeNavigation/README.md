# IDENavigation Components

## Purpose

A collection of common rendering components to enable a consistant look and feel
while navigating CP4I products. The components have individual READMEs on useage
and purpose. The components provided are:

- IdeNavigation - 1st/top level navigation component
- IdeSideNavLink and IdeSideNavMenu - Ide varients of Carbon's SideNavLink and
  SideNavMenu _1_ components
- IdePageHeader - Common page heading component
- IdePageContent - Helper component which wraps the Carbon grid, and provides a
  common way to align CP4I products on the Carbon grid

_1_ SideNavMenu hover state still to be implemented.

## Change history

| Date       | Change                                         | Contributed by / Product         |
| ---------- | ---------------------------------------------- | -------------------------------- |
| 18/10/2019 | Readme update                                  | Matthew Chirgwin / Event Streams |
| 16/09/2019 | First contribution of IdeNavigation components | Matthew Chirgwin / Event Streams |
