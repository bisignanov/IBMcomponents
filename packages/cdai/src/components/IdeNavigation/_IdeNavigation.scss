//
// Copyright IBM Corp. 2020, 2020
//
// This source code is licensed under the Apache-2.0 license found in the
// LICENSE file in the root directory of this source tree.
//
@import '../../globals/scss/carbon-settings';
@import '../../globals/scss/variables';
@import 'carbon-components/scss/components/ui-shell/content';
@import 'carbon-components/scss/components/ui-shell/side-nav';
@import 'carbon-components/scss/components/ui-shell/header';
@import 'carbon-components/scss/components/ui-shell/header-panel';
@import 'carbon-components/scss/components/link/link';
@import 'carbon-components/scss/components/button/button';
@import 'carbon-components/scss/components/tabs/tabs';
@import 'carbon-components/scss/components/breadcrumb/breadcrumb';

@import './PageHeader/IdePageHeader';
@import './PageContent/IdePageContent';
@import './Navigation/IdeNavigation';
