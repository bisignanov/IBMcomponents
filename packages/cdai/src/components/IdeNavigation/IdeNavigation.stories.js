import React from 'react';
import { storiesOf } from '@storybook/react';
import { Header, HeaderName } from 'carbon-components-react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { settings } from 'carbon-components';
import { getComponentLabel } from '../../component_helpers/StorybookHelper';
import styles from './_IdeNavigation.scss';

// import page header nav assets
import {
  withHeadingRenderFunction,
  withHeading,
  withSubtitle,
  withBreadcrumb,
  withTabs,
  withClassName,
  pageHeaderCoreProps,
  bespokeJSXProp,
} from './PageHeader/test_assets/testProps.js';
import IdePageHeaderReadme from './PageHeader/README.md';

// import page content nav assets
import {
  withBleedClass,
  withChildren,
} from './PageContent/test_assets/testProps.js';
import IdePageContentReadMe from './PageContent/README.md';

// import page content nav assets
import {
  navigationCoreProps,
  sampleNavLinks,
  sampleNavMenus,
} from './Navigation/test_assets/testProps.js';
import IdeNavigationReadMe from './Navigation/README.md';

// import top level readme
import topLevelNavigationReadme from './README.md';

// import navigation components
import { IdePageHeader, IdePageContent, IdeNavigation } from '.';

const { prefix } = settings;

const samplePageContent = (
  <IdePageContent>
    <div className={`${prefix}--row`}>
      <span className={`${prefix}--col storybook--idenav-content`}>
        PAGE CONTENT HERE
      </span>
    </div>
  </IdePageContent>
);

const renderMockContent = (component, includeSamplePageContent = true) => (
  <div>
    <Header aria-label="IDE Placeholder heading">
      <HeaderName href="#">IDE Component Library</HeaderName>
    </Header>
    {component}
    {includeSamplePageContent ? samplePageContent : null}
  </div>
);

const interactiveIdeNavigationProps = (props) => {
  return {
    ...props,
    expanded: boolean('Start expanded', true),
    onToggle: action('navigation toggled'),
  };
};

storiesOf(
  `${getComponentLabel('IdeNavigation')}/IDE Navigation component set`,
  module
)
  .addParameters({
    readme: {
      sidebar: topLevelNavigationReadme,
    },
  })
  .addParameters({ styles, layout: 'fullscreen' })
  .add(
    'All components used together',
    () => {
      return renderMockContent(
        <IdeNavigation {...navigationCoreProps}>
          <IdePageHeader {...pageHeaderCoreProps} className={''} />
          {samplePageContent}
        </IdeNavigation>,
        false
      );
    },
    {
      info: {
        text: 'All components in the Navigation pattern used together.',
      },
    }
  );

storiesOf(
  `${getComponentLabel('IdeNavigation')}/IDE Navigation component`,
  module
)
  .addParameters({
    readme: {
      sidebar: IdeNavigationReadMe,
    },
  })
  .addParameters({ styles, layout: 'fullscreen' })
  .addDecorator(withKnobs)
  .add(
    'using IdeSideNavLinks',
    () =>
      renderMockContent(
        <IdeNavigation
          {...navigationCoreProps}
          navigationLinks={sampleNavLinks}>
          {samplePageContent}
        </IdeNavigation>,
        false
      ),
    {
      info: {
        text: 'IdeNavigation with IdeSideNavLink components (one with an icon, one without). Do note, for the blue "active" indicator needs to be provided as either the `isActive` prop, or the `active` attribute on the anchor (react router NavLinks does this for example).',
      },
    }
  )
  .add(
    'Interactive example',
    () =>
      renderMockContent(
        <IdeNavigation {...interactiveIdeNavigationProps(navigationCoreProps)}>
          {samplePageContent}
        </IdeNavigation>,
        false
      ),
    {
      info: {
        text: 'IdeNavigation component with interactive options. See the Knobs for more.',
      },
    }
  );

// seperate so this section can appear in the experimental state
storiesOf(
  `${getComponentLabel('IdeSideNavMenu')}/IDE Navigation component`,
  module
)
  .addParameters({
    readme: {
      sidebar: IdeNavigationReadMe,
    },
  })
  .addParameters({ styles, layout: 'fullscreen' })
  .add(
    'using IdeSideNavMenus',
    () =>
      renderMockContent(
        <IdeNavigation
          {...navigationCoreProps}
          navigationLinks={sampleNavMenus}>
          {samplePageContent}
        </IdeNavigation>,
        false
      ),
    {
      info: {
        text: 'IdeNavigation with IdeSideNavMenu components (one with an icon, one without). Currently these have no alternative hover states as per the guidelines/design. These will be implemented in a future PR.',
      },
    }
  );

storiesOf(
  `${getComponentLabel('IdeNavigation')}/IDE Page header component`,
  module
)
  .addParameters({
    readme: {
      sidebar: IdePageHeaderReadme,
    },
  })
  .addParameters({ styles, layout: 'fullscreen' })
  .add(
    'With breadcrumb, heading, subtitle and tabs properties',
    () =>
      renderMockContent(
        <IdePageHeader {...pageHeaderCoreProps} {...withSubtitle} />
      ),
    {
      info: {
        text: 'Page header component with core common properties having been provided.',
      },
    }
  )
  .add(
    'With just a breadcrumb',
    () =>
      renderMockContent(
        <IdePageHeader {...withBreadcrumb} {...withClassName} />
      ),
    {
      info: {
        text: 'Page header component with just the breadcrumb. Note that no space is reserved for other content',
      },
    }
  )
  .add(
    'With just a heading',
    () =>
      renderMockContent(<IdePageHeader {...withHeading} {...withClassName} />),
    {
      info: {
        text: 'Page header component with just the heading. Note that space is still reserved for the breadcrumb, even if not provided, as well as a small ammount of space below the heading where the tabs would be if provided.',
      },
    }
  )
  .add(
    'With just a heading and subtitle',
    () =>
      renderMockContent(
        <IdePageHeader {...withHeading} {...withSubtitle} {...withClassName} />
      ),
    {
      info: {
        text: 'Page header component with a heading and subtitle. Note that space is still reserved for the breadcrumb, even if not provided, as well as a small ammount of space below the heading where the tabs would be if provided.',
      },
    }
  )
  .add(
    'With a heading and bespoke content alongside the heading',
    () =>
      renderMockContent(
        <IdePageHeader
          {...withHeading}
          {...withHeadingRenderFunction}
          {...withClassName}
        />
      ),
    {
      info: {
        text: 'Page header component with a heading and custom content alongside the heading. This is achieved by render function, which you would provide. It invokes the function with one argument, being the JSX for the heading. Your render function return this heading, along with any other content you want to show, which should include any styling/layout you require. In this instance, we render a Bee icon on the left. For convience, all content in the heading is horizontally aligned.',
      },
    }
  )
  .add(
    'With just tabs',
    () => renderMockContent(<IdePageHeader {...withTabs} {...withClassName} />),
    {
      info: {
        text: 'Page header component with just the tabs. Note that no space is reserved for any other content in this case.',
      },
    }
  )
  .add(
    'With breadcrumb and heading',
    () =>
      renderMockContent(
        <IdePageHeader
          {...withBreadcrumb}
          {...withHeading}
          {...withClassName}
        />
      ),
    {
      info: {
        text: 'Page header component with breadcrumb and heading.',
      },
    }
  )
  .add(
    'With heading and tabs',
    () =>
      renderMockContent(
        <IdePageHeader {...withTabs} {...withHeading} {...withClassName} />
      ),
    {
      info: {
        text: 'Page header component with tabs and heading.',
      },
    }
  )
  .add(
    'With breadcrumb and tabs',
    () =>
      renderMockContent(
        <IdePageHeader {...withBreadcrumb} {...withTabs} {...withClassName} />
      ),
    {
      info: {
        text: 'Page header component with breadcrumb and tabs.',
      },
    }
  )
  .add(
    'With bespoke JSX provided',
    () =>
      renderMockContent(
        <IdePageHeader {...pageHeaderCoreProps} {...bespokeJSXProp} />
      ),
    {
      info: {
        text: 'Page header component with bespoke JSX provided. In this instance, a custom button is rendered as a part of the navigation. This is provided by the customContent prop.',
      },
    }
  );

storiesOf(
  `${getComponentLabel('IdeNavigation')}/IDE Page content component`,
  module
)
  .addParameters({
    readme: {
      sidebar: IdePageContentReadMe,
    },
  })
  .addParameters({ styles, layout: 'fullscreen' })
  .add(
    'With just children',
    () => (
      <div className={'storybook--use-width'}>
        <IdePageContent {...withChildren} />
      </div>
    ),
    {
      info: {
        text: 'Page content component with just children. This component adds a carbon grid wrapping the child, with some IDE specific alignment css. It is recommended you use this to keep content aligned across your/all products.',
      },
    }
  )
  .add(
    'With bleed class set',
    () => (
      <div className={'storybook--use-width'}>
        <IdePageContent {...withBleedClass} {...withChildren} />
      </div>
    ),
    {
      info: {
        text: 'Page content component with children and a bleed class. This allows a bleed effect, larger than the grid. This component adds a carbon grid wrapping the child, with some IDE specific alignment css. It is recommended you use this to keep content aligned across your/all products.',
      },
    }
  );
