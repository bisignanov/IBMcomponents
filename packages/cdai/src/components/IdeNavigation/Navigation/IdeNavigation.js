//
// Copyright IBM Corp. 2020, 2020
//
// This source code is licensed under the Apache-2.0 license found in the
// LICENSE file in the root directory of this source tree.
//

import React from 'react';
import PropTypes from 'prop-types';
import { idAttribute } from '../../../component_helpers/IDHelper';
import {
  SideNav,
  SideNavItems,
  SideNavFooter,
  Content,
} from 'carbon-components-react';
import { idePrefix } from '../../../globals/js/settings';
export default class IdeNavigation extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      expanded: false,
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({ expanded: !this.state.expanded });
  }

  render(
    {
      expanded,
      onToggle,
      showFirstLevel,
      navigationLinks,
      footerDescription,
      ariaAttributes,
      children,
      className,
      contentClassName,
      ...others
    } = this.props
  ) {
    let sideNavClasses = `${idePrefix}-navigation--nav-theme ${idePrefix}-navigation--nav-overrides ${
      !showFirstLevel ? `${idePrefix}-navigation__hide-nav` : ''
    }`;
    let contentClasses = `${idePrefix}-navigation--content ${
      !showFirstLevel ? `${idePrefix}-navigation--content__hide-nav` : ''
    } ${contentClassName}`;

    let toggleFunction =
      typeof onToggle === 'function' && expanded !== undefined
        ? onToggle
        : this.toggle;
    let toggleState = expanded !== undefined ? expanded : this.state.expanded;

    return (
      <div
        {...idAttribute('IdeNavigation')}
        className={`${className} ${idePrefix}-navigation`}
        {...others}
      >
        <SideNav
          {...idAttribute('IdeNavigation--left-nav')}
          {...ariaAttributes}
          className={sideNavClasses}
          isRail
          expanded={toggleState}
          addFocusListeners={false}
          addMouseListeners={false}
        >
          <SideNavItems
            {...idAttribute('IdeNavigation--left-nav-item-container')}
          >
            {navigationLinks}
          </SideNavItems>
          <SideNavFooter
            {...idAttribute('IdeNavigation--left-nav-toggle')}
            className={`${idePrefix}-navigation--toggle`}
            onToggle={toggleFunction}
            expanded={toggleState}
            assistiveText={footerDescription}
          />
        </SideNav>
        <Content
          {...idAttribute('IdeNavigation--content')}
          className={contentClasses}
        >
          {children}
        </Content>
      </div>
    );
  }
}

IdeNavigation.propTypes = {
  /** Required aria attributes. One or the other must be provided */
  ariaAttributes: PropTypes.oneOfType([
    PropTypes.shape({
      /** aria labelledby markup which is applied to the navigation, referencing the label which describe the nav. Must be provided */
      'aria-labelledby': PropTypes.string.isRequired,
    }),
    PropTypes.shape({
      /** aria label which is applied to the navigation. Must be provided, and be pre translated */
      'aria-label': PropTypes.string.isRequired,
    }),
  ]),
  /** children. This should be/represent the current page selected by the user */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
  className: PropTypes.string,
  /** contentClassName - optional styling to be applied to the the container which renders any children */
  contentClassName: PropTypes.string,
  /** expanded. Specify this to make this a controlled component. Use onToggle to be notified of expansion changes */
  expanded: PropTypes.bool,
  /** footerDescription - required. Provide a pre translated string to describe what the expansion control does */
  footerDescription: PropTypes.string.isRequired,
  /** navigation links. An array of Carbon SideNav items, being menus, links, text etc*/
  navigationLinks: PropTypes.arrayOf(PropTypes.node).isRequired,
  /** onToggle. Called when a controlled component to notify the parent of a change in expansion state  */
  onToggle: PropTypes.func,
  /** showFirstLevel. Defaults to true. Set this to false to only show the content, and not the left navigation */
  showFirstLevel: PropTypes.bool,
  /** className - optional styling to be applied to the component */
};

IdeNavigation.defaultProps = {
  showFirstLevel: true,
  className: '',
  contentClassName: '',
  footerDescription:
    'Click here to expand or collapse the navigation - change this for a pre translated string!',
  navigationLinks: [],
  ariaAttributes: {
    'aria-label': 'Example Aria Labeled',
  },
};
