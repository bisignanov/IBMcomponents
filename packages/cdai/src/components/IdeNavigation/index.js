import IdePageHeader from './PageHeader/IdePageHeader';
import IdePageContent from './PageContent/IdePageContent';
import IdeNavigation from './Navigation/IdeNavigation';
import {
  IdeSideNavLink,
  IdeSideNavMenu,
} from './Navigation/IdeNavigationItems';
export {
  IdePageHeader,
  IdePageContent,
  IdeNavigation,
  IdeSideNavLink,
  IdeSideNavMenu,
};
