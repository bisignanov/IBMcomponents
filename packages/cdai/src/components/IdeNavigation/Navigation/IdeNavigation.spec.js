//
// Copyright IBM Corp. 2020, 2020
//
// This source code is licensed under the Apache-2.0 license found in the
// LICENSE file in the root directory of this source tree.
//

import IdeNavigation from './IdeNavigation';
import { navigationCoreProps } from './test_assets/testProps.js';
import { idAttributeSelector } from '../../../component_helpers/IDHelper';
import * as jth from '../../../component_helpers/jest_test_helper_functions.js';

describe('IdeNavigation unit tests', () => {
  let component, unmount;

  const mountTestComponent = (defaultProps = {}, props = {}) => {
    let test = jth.mountComponent(
      jth.getJSXForComponent(IdeNavigation, defaultProps, props)
    );
    component = test.component;
    unmount = test.unmount;
  };

  afterEach(() => {
    // if unmount defined, call it to clear up any mounted component
    unmount && unmount();
  });

  describe('view tests', () => {
    it('renders the expected structure with no props provided', () => {
      mountTestComponent();
      expect(component.getElements()).toMatchSnapshot();
    });

    it('renders the expected structure with all provided', () => {
      mountTestComponent({ ...navigationCoreProps });
      expect(component.getElements()).toMatchSnapshot();
    });

    it('adds the expected classes when the side nav is hidden via prop', () => {
      mountTestComponent({ showFirstLevel: false });
      expect(
        component
          .find(idAttributeSelector('IdeNavigation--left-nav'))
          .hasClass('ide-navigation__hide-nav')
      ).toBe(true);
      expect(
        component
          .find(idAttributeSelector('IdeNavigation--content'))
          .hasClass('ide-navigation--content__hide-nav')
      ).toBe(true);
    });
  });

  describe('function tests', () => {
    it('toggle() when invoked flips the expanded state', () => {
      mountTestComponent();
      const componentInstance = component.instance();
      let startingExpandedState = componentInstance.state.expanded;
      componentInstance.toggle();
      expect(componentInstance.state.expanded).toBe(!startingExpandedState);
    });

    it('if controlled, the parent component dictates expanded state, and gets called on navigation interaction', () => {
      const mockToggle = jest.fn();
      mountTestComponent({ expanded: true, onToggle: mockToggle });
      expect(
        component.find(idAttributeSelector('IdeNavigation--left-nav')).props()
          .expanded
      ).toBe(true); // as we have provided it
      component
        .find(idAttributeSelector('IdeNavigation--left-nav-toggle'))
        .props()
        .onToggle(); // trigger the callback
      expect(mockToggle).toHaveBeenCalled(); // and confirm our callback is invoked on toggle
    });
  });
});
