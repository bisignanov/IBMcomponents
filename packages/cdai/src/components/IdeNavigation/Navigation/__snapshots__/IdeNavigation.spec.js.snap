// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeNavigation unit tests view tests renders the expected structure with all provided 1`] = `
Array [
  <div
    className=" ide-navigation"
    data-ui-id="IdeNavigation"
  >
    <SideNav
      addFocusListeners={false}
      addMouseListeners={false}
      aria-label="sample aria label to describe the nav"
      aria-labelledby="#myId"
      className="ide-navigation--nav-theme ide-navigation--nav-overrides "
      data-ui-id="IdeNavigation--left-nav"
      defaultExpanded={false}
      expanded={false}
      isChildOfHeader={true}
      isFixedNav={false}
      isPersistent={true}
      isRail={true}
    >
      <SideNavItems
        data-ui-id="IdeNavigation--left-nav-item-container"
      >
        <IdeSideNavMenu
          title="Sample menu"
        >
          <SideNavMenuItem
            href="javascript:void(0)"
          >
            Menu item one
          </SideNavMenuItem>
          <SideNavMenuItem
            href="javascript:void(0)"
          >
            Menu item two
          </SideNavMenuItem>
          <SideNavMenuItem
            href="javascript:void(0)"
          >
            Menu item three
          </SideNavMenuItem>
        </IdeSideNavMenu>
        <IdeSideNavMenu
          renderIcon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          title="Sample menu with icon"
        >
          <SideNavMenuItem
            href="javascript:void(0)"
          >
            Menu item one
          </SideNavMenuItem>
          <SideNavMenuItem
            href="javascript:void(0)"
          >
            Menu item two
          </SideNavMenuItem>
          <SideNavMenuItem
            href="javascript:void(0)"
          >
            Menu item three
          </SideNavMenuItem>
        </IdeSideNavMenu>
        <IdeSideNavLink
          href="javascript:void()"
          renderIcon={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
        >
          Sample Link with Icon
        </IdeSideNavLink>
        <IdeSideNavLink
          href="javascript:void()"
        >
          Sample Link
        </IdeSideNavLink>
      </SideNavItems>
      <SideNavFooter
        assistiveText="Expand or collapse the navigation"
        className="ide-navigation--toggle"
        data-ui-id="IdeNavigation--left-nav-toggle"
        expanded={false}
        onToggle={[Function]}
      />
    </SideNav>
    <Content
      className="ide-navigation--content  "
      data-ui-id="IdeNavigation--content"
      tagName="main"
    />
  </div>,
]
`;

exports[`IdeNavigation unit tests view tests renders the expected structure with no props provided 1`] = `
Array [
  <div
    className=" ide-navigation"
    data-ui-id="IdeNavigation"
  >
    <SideNav
      addFocusListeners={false}
      addMouseListeners={false}
      aria-label="Example Aria Labeled"
      className="ide-navigation--nav-theme ide-navigation--nav-overrides "
      data-ui-id="IdeNavigation--left-nav"
      defaultExpanded={false}
      expanded={false}
      isChildOfHeader={true}
      isFixedNav={false}
      isPersistent={true}
      isRail={true}
    >
      <SideNavItems
        data-ui-id="IdeNavigation--left-nav-item-container"
      />
      <SideNavFooter
        assistiveText="Click here to expand or collapse the navigation - change this for a pre translated string!"
        className="ide-navigation--toggle"
        data-ui-id="IdeNavigation--left-nav-toggle"
        expanded={false}
        onToggle={[Function]}
      />
    </SideNav>
    <Content
      className="ide-navigation--content  "
      data-ui-id="IdeNavigation--content"
      tagName="main"
    />
  </div>,
]
`;
